### PR TITLE
TYP: Narrow down type of method keyword in Index

### PR DESCRIPTION
**Pull request by kklein**
This PR narrows down the type of the `method` keyword, used in two methods of the class `Index` to exactly the permitted values via the use of `Literal`.

- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Comment by kklein** [comment]
@phofl @noatamir @jorisvandenbossche @MarcoGorelli PyData workshop

File: pandas/core/indexes/base.py [hunk 1]
```diff
@@ -208,6 +208,7 @@
 _index_shared_docs: dict[str, str] = {}
 str_t = str
 
+IndexMethod = Literal["pad", "ffill", "backfill", "bfill", "nearest"]
```
Comment (twoertwein) [inline pandas/core/indexes/base.py:211]: This should go in `pandas/_typing.py`

**Comment by kklein** [comment]
Any suggestions on how to proceed with `clean_reindex_fill_method` and `clean_fill_method`?

**Review by twoertwein — APPROVED** [review]

**Comment by twoertwein** [comment]
Didn't see that mypy failed:

> pandas/core/indexes/base.py:3749: error: Incompatible types in assignment (expression has type "Optional[str]", variable has type "Optional[Literal['pad', 'ffill', 'backfill', 'bfill', 'nearest']]")  [assignment]

You might be able to change the signature of `clean_reindex_fill_method` or add an ignore comment.

**Comment by phofl** [comment]
Yep you can change the signature as well

**Comment by kklein** [comment]
@phofl @twoertwein 
Sorry about the delay!

The mypy/typing problem seems to be resolved, yet some unit tests are failing. Could it be that those are flaky?

**Comment by twoertwein** [comment]
Thank you @kklein !

**Pull request merged**